Xledger.Sql/Collections/ImmList.cs

```diff
@@ -119,10 +119,8 @@ public bool Equals(ImmList<T> other) {
 
         public override int GetHashCode() {
             var h = 17;
-            foreach (var e in data) {
-                if (e == null) {
-                    h = h * 23 + e.GetHashCode();
-                }
+            for (int i = 0; i < data.Length; i++) {
+                h = h * 23 + data[i]?.GetHashCode() ?? 0;

```

Comment:
```suggestion
                h = h * 23 + (data[i]?.GetHashCode() ?? 0);
```

The precedence of `??` is low enough that it ends up returning 0 and "erasing" h if there's a null in the list.

Reply:
Oh, scary